media/src/main/java/io/wcm/handler/media/ui/ResourceMedia.java

```diff
@@ -70,8 +82,55 @@
   @RequestAttribute(injectionStrategy = InjectionStrategy.OPTIONAL)
   private String cssClass;
 
+  /**
+   * Defines responsive rendition widths for image.
+   * To be used together with 'imageSizes' property. <br>
+   * Example: "{@literal 2560:false,1920:false,1280:false,640:false,320:false}" <br>
+   * The boolean flag indicates whether the with is required.<br>
+   * Cannot be used together with the picture source parameters.
+   */
+  @RequestAttribute(injectionStrategy = InjectionStrategy.OPTIONAL)
+  private String widthOptions;
+
+  /**
+   * "Sizes" string for img element.
+   * Example: "{@literal (min-width: 400px) 400px, 100vw}"<br>
+   * Cannot be used together with the picture source parameters.
+   */
+  @RequestAttribute(injectionStrategy = InjectionStrategy.OPTIONAL)
+  private String imageSizes;
+
+  /**
+   * List of media formats for the picture source elements.
+   * Example: "{@literal ['mf_16_9']}"<br>
+   * You have to define the same number of array items in all pictureSource* properties.
+   * Cannot be used together with image sizes.
+   */
+  @RequestAttribute(injectionStrategy = InjectionStrategy.OPTIONAL)
+  private Object[] pictureSourceMediaFormat;
+
+  /**
+   * List of media expressions for the picture source elements.
+   * Example: "{@literal ['(max-width: 799px)', '(min-width: 800px)']}"<br>
+   * You have to define the same number of array items in all pictureSource* properties.
+   * Cannot be used together with image sizes.
+   */
+  @RequestAttribute(injectionStrategy = InjectionStrategy.OPTIONAL)
+  private Object[] pictureSourceMedia;
+
+  /**
+   * List of widths for the picture source elements.
+   * Example: "{@literal 479,719,959,1279,1439,1440}"<br>
```

Comment:
add info about support for ? suffixed for the width to mark an with as optional - see
https://wcm.io/handler/media/component-properties.html